### PR TITLE
feat: enforce `type` field of a cellbase output cell must be absent

### PR DESCRIPTION
**Pull request by zhangsoledad**
* remove redundant EmptyVerifier
* enforce `type` field of a cellbase output cell must be absent

### CODE REVIEW

File: verification/src/block_verifier.rs [hunk 1]
```diff
@@ -99,6 +90,16 @@ impl<CP: ChainProvider + Clone> CellbaseVerifier<CP> {
         {
             return Err(Error::Cellbase(CellbaseError::InvalidInput));
         }
+
+        // currently, we enforce`type` field of a cellbase output cell must be absent
```
Comment (janx) [inline verification/src/block_verifier.rs]: why do we enforce this?

Comment (zhangsoledad) [reply verification/src/block_verifier.rs]: Currently, we have an involuntary issue that CKB-VM execution skipped `cellbase`. It's a temporary solution until we implement `cellbase` type script verifiction.

File: verification/src/block_verifier.rs [hunk 2]
```diff
@@ -75,9 +69,6 @@ impl<CP: ChainProvider + Clone> CellbaseVerifier<CP> {
     }
 
     pub fn verify(&self, block: &Block) -> Result<(), Error> {
-        if block.commit_transactions().is_empty() {
```
Comment (doitian) [inline verification/src/block_verifier.rs]: Is cellbase required now? Has it been added in the verification RFC doc?

File: verification/src/block_verifier.rs (hunk 1 above)
Comment (janx) [reply verification/src/block_verifier.rs]: Feels like something we should fix before testnet.

File: verification/src/block_verifier.rs (hunk 2 above)
Comment (zhangsoledad) [reply verification/src/block_verifier.rs]: Cellbase is always required. RFC updated .

**Pull request merged**